### PR TITLE
RSDK-2866: add disconnection and reconnection events

### PR DESCRIPTION
**Pull request by purplenicole730**

### CODE REVIEW

File: src/robot/robot.ts [hunk 1]
```diff
@@ -146,4 +148,15 @@ export interface Robot {
...
   ): RobotStatusStream;
+
+  /**
+   * Call a function when an event of either 'reconnected' or 'disconnected' is 
+   * triggered.
+   * 
+   * @param event - The event that was triggered.
+   * @param listener - The function to call
+   * @group Connection
+   * @alpha
```
Comment (purplenicole730) [inline src/robot/robot.ts:159]: What `@group` should this be in, if any? What does `@alpha` do?

File: src/robot/client.ts [hunk 2]
```diff
@@ -353,32 +389,7 @@ export class RobotClient implements Robot {
...
-                  console.debug(
-                    `failed to reconnect - retries count: ${retries}`
-                  );
-                  retries += 1;
-                  throw error;
-                }
-              )
-            );
-          }
+          this.isConnected() ? events.emit('reconnected', {}) : events.emit('disconnected', {});
```
Comment (purplenicole730) [inline src/robot/client.ts:392]: Lint error caused by this line:
`392:11  error  Expected an assignment or function call and instead saw an expression`

File: src/robot/robot.ts (hunk 1 above)
Comment (maximpertsov) [reply src/robot/robot.ts:159]: the group is purely for organization -  we can change this later or even leave it off. `Event Listeners` might be a good name if we end up adding non-connection events later.

File: src/robot/robot.ts [hunk 3]
```diff
@@ -146,4 +148,15 @@ export interface Robot {
     resourceNames?: ResourceName.AsObject[],
     durationMs?: number
   ): RobotStatusStream;
+
+  /**
+   * Call a function when an event of either 'reconnected' or 'disconnected' is 
+   * triggered.
+   * 
+   * @param event - The event that was triggered.
```
Comment (maximpertsov) [inline src/robot/robot.ts:156]: [nit] we might want to note the event types here.

Separately, we might also want to call out that these events only apply to WebRTC connections.

**Review by maximpertsov — CHANGES_REQUESTED** [review]
looks great! some minor suggestions

File: src/robot/robot.ts [hunk 4]
```diff
@@ -146,4 +149,19 @@ export interface Robot {
     resourceNames?: ResourceName.AsObject[],
     durationMs?: number
   ): RobotStatusStream;
+
+  /**
+   * Call a function when an event of either 'reconnected' or 'disconnected' is
+   * triggered. NOTE: Please note that these functions currently only apply to
+   * WebRTC connections.
```
Comment (maximpertsov) [inline src/robot/robot.ts:156]:
```suggestion
   * triggered. Note that these events will only be triggered on WebRTC connections.
```

**Review by maximpertsov — APPROVED** [review]

**Pull request merged**